File: typescript/src/payment/application/policies/transactionRefund.ts
// Policy che traduce RefundRequested in RefundTransaction.

import type { Policy } from '@/common/application/policy.js';
import { RefundTransaction, refundTransaction } from '../../application/commands/refundTransaction.js';
import { RefundRequested } from '../../domain/events/refundRequested.js';
import { requireArgument } from '@/common/utils/requireArgument.js';

export class TransactionRefund implements Policy<RefundRequested, RefundTransaction> {
  evaluate(event: RefundRequested): RefundTransaction {
    requireArgument(event, 'event');
    return refundTransaction(event.aggregateId, event.amount);
  }
}

File: java/src/main/java/it/giannibombelli/wsc2026/payment/application/policies/TransactionRefund.java
package it.giannibombelli.wsc2026.payment.application.policies;

import it.giannibombelli.wsc2026.common.utils.Require;

import it.giannibombelli.wsc2026.common.application.Policy;
import it.giannibombelli.wsc2026.payment.application.commands.RefundTransaction;
import it.giannibombelli.wsc2026.payment.domain.events.RefundRequested;

public final class TransactionRefund implements Policy<RefundRequested, RefundTransaction> {

    @Override
    public RefundTransaction evaluate(RefundRequested event) {
        Require.requireArgument(event, "event");
        return new RefundTransaction(event.aggregateId(), event.amount());
    }
}
